src/Domain/FixerFileProcessor.php

```diff
@@ -0,0 +1,190 @@
+
+    public function __construct(
+        CachedFileLoader $cachedFileLoader,
+        ErrorAndDiffCollector $errorAndDiffCollector,
+        FileToTokensParser $fileToTokensParser,
+        Differ $differ
+    ) {
+        $this->cachedFileLoader = $cachedFileLoader;
+        $this->errorAndDiffCollector = $errorAndDiffCollector;
+        $this->fileToTokensParser = $fileToTokensParser;
+        $this->differ = $differ;
+    }
+
+    public function addChecker(FixerInterface $fixer): void
+    {
+        $this->fixers[] = $fixer;
+    }
+
+    /**
+     * @return array<\PhpCsFixer\Fixer\FixerInterface>
+     */
+    public function getCheckers(): array
+    {
+        return $this->fixers;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function processFile(SmartFileInfo $smartFileInfo): string
+    {
+        $filePath = $smartFileInfo->getRealPath();
+        if ($filePath === false) {
+            throw new \LogicException('Unable to found file ' . $smartFileInfo->getFilename());
+        }
+        $oldContent = $this->cachedFileLoader->getFileContent($smartFileInfo);
+        $tokens = $this->fileToTokensParser->parseFromFilePath($filePath);
+
+        /** @var FixerInterface $fixer */
+        foreach ($this->getCheckers() as $fixer) {
+            $this->processTokensByFixer($smartFileInfo, $tokens, $fixer);
+            if (! $tokens->isChanged()) {
+                continue;
+            }
+
+            $newContent = $tokens->generateCode();
+            $fullDiff = $this->differ->diff($oldContent, $newContent);
+
+            $this->processDiff($fullDiff, get_class($fixer), $smartFileInfo);
+
+            Tokens::clearCache();
+            // Reinit tokens
+            $tokens = $this->fileToTokensParser->parseFromFilePath($filePath);
+        }
+
+        return 'processed';
+    }
+
+    private function processTokensByFixer(SmartFileInfo $smartFileInfo, Tokens $tokens, FixerInterface $fixer): void
+    {
+        try {
+            $fixer->fix($smartFileInfo, $tokens);
+        } catch (\Throwable $throwable) {
+            throw new FixerFailedException(sprintf(
+                'Fixing of "%s" file by "%s" failed: %s in file %s on line %d',
+                $smartFileInfo->getRelativeFilePath(),
+                get_class($fixer),
+                $throwable->getMessage(),
+                $throwable->getFile(),
+                $throwable->getLine()
+            ), $throwable->getCode(), $throwable);
+        }
+
+        if (! $tokens->isChanged()) {
+            return;
+        }
+    }
+
+    private function processDiff(string $diff, string $fixerClass, SmartFileInfo $smartFileInfo): void
+    {
+        $parsedDiff = $this->splitStringByLines($diff);
+        // Get first line number & Remove headers of diff
+        $currentLineNumber = $this->parseLineNumber($parsedDiff[2]);
+        $parsedDiff = array_slice($parsedDiff, 3);
+
+        $headerMessage = "You should change following \n";
```

Comment:
If you have any ideas for this message, I'm interested :wink:

Reply:
Sounds good to me tbh. :)